### PR TITLE
Resolve womtool errors and kmer consistency

### LINKED ISSUE #61: Implement kmer consistency
kmer consistency check was never implemented in the WDL workflow, so none of the modimer outputs are used. 

For reference:
https://github.com/PacificBiosciences/pb-human-wgs-workflow-snakemake/blob/main/rules/sample_kmer_consistency.smk

This is a pairwise comparison of the modimers.tsv files from each movie, after subtracting the reference modimers.tsv. Essentially:

movie1 modimers - reference modimers -> non-reference movie1 modimers
movie2 modimers - reference modimers -> non-reference movie2 modimers

count(subtract(non-reference movie1 modimers, non-reference movie2 modimers)) + count(subtract(non-reference movie2 modimers, non-reference movie1 modimers)) -> count of unique modimers from pairwise comparison

We report a metric representing the proportion of total non-reference modimers that are unique to one movie. If this count is above some threshold, it's likely that the two movies (SMRT Cells) were loaded with different samples.

### LINKED ISSUE #67: hifiasm still uses aligned bam instead of uBAM
Bug in pbbm2 that doesn't have all the unaligned bams. So using original ubams

### LINKED ISSUE #69: Add documentation for how to change URLS
Documentation to how to change the urls using script.

### LINKED ISSUE #70: Update documentation for reference, resources and test data

### PR DESCRIPTION
**Pull request by vsmalladi**
Fix outstanding womtool errors.

Also Implement kmer consistency that was missing. Closes #61 
Fixes logic for ubam use in hifiasm Closes #67 
Add documentation for how to change URLS Closes #69 
Update documentation for reference, resources and test data Closes #70

### CODE REVIEW

**Comment by vsmalladi** [comment]
@byoo This should all validate now. Had to do some weird wdl manipulation to get it to work.

@williamrowell also implemented the kmer consistency since it was not there before. Best to have this for 1.0 release.

**Comment by vsmalladi** [comment]
@cbi-star, @byoo, @williamrowell @ducatiMonster916 Ready for code review. Fixes most outstanding issues with womtool and logic for ubam.

**Review by williamrowell — APPROVED** [review]
LGTM

Edit: This was referring to a small subset of the changes.  Didn't see the others at the time.

File: rename_urls.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@
 remote_user="PacificBiosciences"
 remote_repo="pb-human-wgs-workflow-wdl"
 
-local_user="TimD1"
+local_user="user"
```
Comment (williamrowell) [inline rename_urls.sh]: Perhaps `local_user=$(whoami)`?

File: rename_urls.sh [hunk 2]
```diff
@@ -19,13 +19,13 @@ echo " "
 if [ $# -eq 1 ] && [ "$1" == "--remote" ]; then
 	echo -en "Changing all WDL Github URLs from '${local_user}/${local_repo}' to '${remote_user}/${remote_repo}'..."
 	find . -type f -name "*.wdl" -print0 | \
-		xargs -0 sed -i "s/com\/${local_user}\/${local_repo}/com\/${remote_user}\/${remote_repo}/g"
+		xargs -0 gsed -i "s/com\/${local_user}\/${local_repo}/com\/${remote_user}\/${remote_repo}/g"
```
Comment (williamrowell) [inline rename_urls.sh]: Correct me if I'm wrong, but isn't `gsed` FreeBSD/macOS specific?

File: workflows/reference/reference_build.wdl [hunk 3]
```diff
@@ -1,22 +1,26 @@
 version 1.0
 
-import "https://raw.githubusercontent.com/PacificBiosciences/pb-human-wgs-workflow-wdl/main/workflows/reference/tasks/clinvar.wdl" as clinvar
-import "https://raw.githubusercontent.com/PacificBiosciences/pb-human-wgs-workflow-wdl/main/workflows/reference/tasks/ensembl.wdl" as ensembl
-import "https://raw.githubusercontent.com/PacificBiosciences/pb-human-wgs-workflow-wdl/main/workflows/reference/tasks/lof.wdl" as lof
+#import "./tasks/clinvar.wdl" as clinvar
```
Comment (williamrowell) [inline workflows/reference/reference_build.wdl:3]: In other WDLs, you uncomment the absolute URIs and comment the relative URIs.  Here you comment out both.  Is this intentional?

**Review by williamrowell — COMMENTED** [review]
A few open questions.

File: rename_urls.sh (hunk 1 above)
Comment (vsmalladi) [reply rename_urls.sh]: Yes good idea

File: rename_urls.sh (hunk 2 above)
Comment (vsmalladi) [reply rename_urls.sh]: Yes I will update

File: workflows/reference/reference_build.wdl (hunk 3 above)
Comment (vsmalladi) [reply workflows/reference/reference_build.wdl:3]: Nope accident part of testing

File: rename_urls.sh [hunk 4]
```diff
@@ -19,13 +19,13 @@ echo " "
...
 	find . -type f -name "*.wdl" -print0 | \
-		xargs -0 gsed -i "s/com\/${local_user}\/${local_repo}/com\/${remote_user}\/${remote_repo}/g"
+		xargs -0 sed -i "s/com\/${local_user}\/${local_repo}/com\/${remote_user}\/${remote_repo}/g"
 	echo "done!"
 
 elif [ $# -eq 1 ] && [ "$1" == "--local" ]; then
 	echo -en "Changing all WDL Github URLs from '${remote_user}/${remote_repo}' to '${local_user}/${local_repo}'..."
 	find . -type f -name "*.wdl" -print0 | \
-		xargs -0 gsed -i "s/com\/${remote_user}\/${remote_repo}/com\/${local_user}\/${local_repo}/g"
+		xargs -0 see -i "s/com\/${remote_user}\/${remote_repo}/com\/${local_user}\/${local_repo}/g"
```
Comment (williamrowell) [inline rename_urls.sh]: see -> sed

**Review by williamrowell — APPROVED** [review]

**Comment by williamrowell** [comment]
Ready to test. @byoo

**Comment by byoo** [comment]
@vsmalladi, I had validation errors using womtool 73. Could you take a look?

> Failed to import 'https://raw.githubusercontent.com/PacificBiosciences/pb-human-wgs-workflow-wdl/main/workflows/sample/sample.trial.wdl' (reason 1 of 1): Failed to import 'https://raw.githubusercontent.com/PacificBiosciences/pb-human-wgs-workflow-wdl/main/workflows/sample/sample.wdl' (reason 1 of 1): Failed to import 'https://raw.githubusercontent.com/PacificBiosciences/pb-human-wgs-workflow-wdl/main/workflows/sample/tasks/check_kmer_consistency.wdl' (reason 1 of 1): Failed to process task definition 'check_kmer_consistency' (reason 1 of 1): This workflow contains a cyclic dependency:
> "File check_kmer_consistency_output_name = ~{__check_kmer_consistency_output_name}" -> "String __check_kmer_consistency_output_name = select_first([check_kmer_consistency_output_name, ~{sample_name}.kmerconsistency.txt])"
> "String __check_kmer_consistency_output_name = select_first([check_kmer_consistency_output_name, ~{sample_name}.kmerconsistency.txt])" -> "File check_kmer_consistency_output_name = ~{__check_kmer_consistency_output_name}"
> Failed to import 'https://raw.githubusercontent.com/PacificBiosciences/pb-human-wgs-workflow-wdl/main/workflows/sample/sample_trio.trial.wdl' (reason 1 of 1): Failed to import 'https://raw.githubusercontent.com/PacificBiosciences/pb-human-wgs-workflow-wdl/main/workflows/sample/sample.wdl' (reason 1 of 1): Failed to import 'https://raw.githubusercontent.com/PacificBiosciences/pb-human-wgs-workflow-wdl/main/workflows/sample/tasks/check_kmer_consistency.wdl' (reason 1 of 1): Failed to process task definition 'check_kmer_consistency' (reason 1 of 1): This workflow contains a cyclic dependency:
> "File check_kmer_consistency_output_name = ~{__check_kmer_consistency_output_name}" -> "String __check_kmer_consistency_output_name = select_first([check_kmer_consistency_output_name, ~{sample_name}.kmerconsistency.txt])"
> "String __check_kmer_consistency_output_name = select_first([check_kmer_consistency_output_name, ~{sample_name}.kmerconsistency.txt])" -> "File check_kmer_consistency_output_name = ~{__check_kmer_consistency_output_name}"

**Comment by vsmalladi** [comment]
@byoo interesting womtool 80 validates and 73 fails on this step. Working on fixing.

**Pull request merged**